test/integration/getserverprops/test/index.test.js

```diff
@@ -0,0 +1,373 @@
+    ),
+  },
+  '/': {
+    page: '/',
+    dataRouteRegex: normalizeRegEx(
+      `^\\/_next\\/data\\/${escapeRegex(buildId)}\\/index.json$`
+    ),
+  },
+  '/default-revalidate': {
+    page: '/default-revalidate',
+    dataRouteRegex: normalizeRegEx(
+      `^\\/_next\\/data\\/${escapeRegex(buildId)}\\/default-revalidate.json$`
+    ),
+  },
+  '/catchall/[...path]': {
+    page: '/catchall/[...path]',
+    dataRouteRegex: normalizeRegEx(
+      `^\\/_next\\/data\\/${escapeRegex(buildId)}\\/catchall\\/(.+?)\\.json$`
+    ),
+  },
+  '/blog': {
+    page: '/blog',
+    dataRouteRegex: normalizeRegEx(
+      `^\\/_next\\/data\\/${escapeRegex(buildId)}\\/blog.json$`
+    ),
+  },
+  '/blog/[post]/[comment]': {
+    page: '/blog/[post]/[comment]',
+    dataRouteRegex: normalizeRegEx(
+      `^\\/_next\\/data\\/${escapeRegex(
+        buildId
+      )}\\/blog\\/([^/]+?)\\/([^/]+?)\\.json$`
+    ),
+  },
+  '/user/[user]/profile': {
+    page: '/user/[user]/profile',
+    dataRouteRegex: normalizeRegEx(
+      `^\\/_next\\/data\\/${escapeRegex(
+        buildId
+      )}\\/user\\/([^/]+?)\\/profile\\.json$`
+    ),
+  },
+  '/another': {
+    page: '/another',
+    dataRouteRegex: normalizeRegEx(
+      `^\\/_next\\/data\\/${escapeRegex(buildId)}\\/another.json$`
+    ),
+  },
+})
+
+const navigateTest = (dev = false) => {
+  it('should navigate between pages successfully', async () => {
+    const toBuild = [
+      '/',
+      '/another',
+      '/something',
+      '/normal',
+      '/blog/post-1',
+      '/blog/post-1/comment-1',
+    ]
+
+    await Promise.all(toBuild.map(pg => renderViaHTTP(appPort, pg)))
+
+    const browser = await webdriver(appPort, '/')
+    let text = await browser.elementByCss('p').text()
+    expect(text).toMatch(/hello.*?world/)
+
+    // hydration
+    await waitFor(2500)
+
+    // go to /another
+    async function goFromHomeToAnother() {
+      await browser.elementByCss('#another').click()
+      await browser.waitForElementByCss('#home')
+      text = await browser.elementByCss('p').text()
+      expect(text).toMatch(/hello.*?world/)
+    }
+    await goFromHomeToAnother()
+
+    // go to /
+    async function goFromAnotherToHome() {
+      await browser.eval('window.didTransition = 1')
+      await browser.elementByCss('#home').click()
+      await browser.waitForElementByCss('#another')
+      text = await browser.elementByCss('p').text()
+      expect(text).toMatch(/hello.*?world/)
+      expect(await browser.eval('window.didTransition')).toBe(1)
+    }
+    await goFromAnotherToHome()
+
+    await goFromHomeToAnother()
+    const snapTime = await browser.elementByCss('#anotherTime').text()
+
+    // Re-visit page
+    await goFromAnotherToHome()
+    await goFromHomeToAnother()
+
+    const nextTime = await browser.elementByCss('#anotherTime').text()
+    if (dev) {
+      expect(snapTime).not.toMatch(nextTime)
+    } else {
+      expect(snapTime).toMatch(nextTime)
```

Comment:
Server props should always fetch fresh data on navigation, correct? If anything, we'd want to skip it on back but maybe re-fetch it on forward nav.

Reply:
I updated it to never cache for `getServerProps` since it doesn't seem like you'd want any caching if using it over `getStaticProps`. We can investigate caching for history navigations in follow-ups if we think it is desirable